### DISCUSSION TITLE
[BUG] Key issue

### DISCUSSION THREAD

**Discussion by imdkbj** [MobileReality/react-native-select-pro#216] [General]
**Describe the bug**
Can't use two list with same value, getting error of unique key.

**Expected behaviour**
Key should not depend on the value prop. 
There should be option for custom value, label like as react select.

```js
  <Select
                          isDisabled={!stateId}
                          options={citiesArray}
                          classNamePrefix="select"
                          getOptionValue={({ id }) => id}
                          getOptionLabel={({ name }) => name}
                          theme={selectThemeColors}
                          className={'react-select'}
                          {...field}
                        />
```

**Comment by kkaplinski** [comment]
@imdkbj Can you explain your use case ? You want to have two options on the list with the same value, or you want to have two select lists with the same options ?

**Comment by imdkbj** [comment]
@kkaplinski Thanks for the quick response.Let me explain the use case.

#### We will get data from server in following format.
```js
const data1 = [
    { id: 1, name: 'Name1' },
    { id: 2, name: 'Name2' },
]

const data2 = [
    { id: 1, name: 'X' },
    { id: 2, name: 'Y' },
]
```

These will be assigned to two different Select list.

So, We need to manually process the array to convert in label, value key of the objects.
~This will show error for two children with the same key.~
~Will also throw - measure cannot find view with tag #2687~

_A bonus issue : Dynamic option array will not update. ( I have a hook which will fetch data from server but found that option list not updated with the data. I have to trigger the component via a key change.)_

**Comment by kkaplinski** [comment]
Hello @imdkbj. Thank you for the comments. Two lists with the same values/keys should not be a problem - you will have two separate components.
Regarding to loading options async: We are not supporting currently this possibility, but we will think about it. For now you can fetch data from server and show select component after new data arrived.

**Comment by imdkbj** [comment]
Hmm, Thanks for the info.
